lib/tasks/db.rake

```diff
@@ -16,4 +16,135 @@ namespace :db do
       end
     end
   end
+
+  # Before we load the schema, define the timestamp_id function.
+  # Idiomatically, we might do this in a migration, but then it
+  # wouldn't end up in schema.rb, so we'd need to figure out a way to
+  # get it in before doing db:setup as well. This is simpler, and
+  # ensures it's always in place.
+  Rake::Task['db:schema:load'].enhance ['db:define_timestamp_id']
```

Comment:
The other solution is changing schema format from rb to sql

Reply:
Personally I'd prefer to keep schema.rb and deal with this rather than shift to the less-used SQL format, but I don't care strongly.